libyul/Object.h

```diff
@@ -66,9 +66,12 @@ struct Object: ObjectNode
 	/// @returns a (parseable) string representation. Includes types if @a _yul is set.
 	std::string toString(Dialect const* _dialect) const override;
 
+	void addNamedSubObject(YulString _name, std::shared_ptr<ObjectNode> _subObject);
+
 	/// @returns the set of names of data objects accessible from within the code of
 	/// this object.
 	std::set<YulString> dataNames() const;
```

Comment:
https://github.com/ethereum/solidity/pull/9047#discussion_r431790538

This PR completely changes semantics of `dataNames()`. It looks like earlier it would return the name of the object and the names of subobjects, all in a single set (e.g. `{"A", "B", "C"}`). Now it returns whole "paths" (e.g. `{"A", "A.B", "A.C", "A.C.D.E"}`) which means that the names are different and that there can be more of them than before.

I think this warrants a name change. I'd call it `qualifiedDataNames()` or something like that.

Also, I think that the docstring should explicitly say that the names are included recursively and qualified with subobject names. Maybe even it should have a short, one-line example.